lib/orderbook/OrderBook.ts

```diff
@@ -532,37 +532,40 @@ class OrderBook extends EventEmitter {
     const { r_hash, proposedQuantity, orderId, pairId } = requestPacket.body!;
 
     const order = this.tryGetOwnOrder(orderId, pairId);
-    if (order) {
-      const availableQuantity = order.hold ? order.quantity - order.hold : order.quantity;
-      // TODO: accept the smaller of the proposed quantity and the available quantity
-      if (availableQuantity >= proposedQuantity) {
-        // put accepted quantity on hold
-        // quantityToAccept = Math.min(proposedQuantity, availableQuantity);
-        const quantityToAccept = proposedQuantity;
-
-        order.hold = order.hold ? order.hold + quantityToAccept : quantityToAccept;
-
-        // try to accept the deal
-        const orderToAccept = {
-          quantityToAccept,
-          localId: order.localId,
-          price: order.price,
-        };
-        const dealAccepted = await this.swaps!.acceptDeal(orderToAccept, requestPacket, peer);
-        if (!dealAccepted) {
-          // release hold amount and reject swap
-          order.hold -= quantityToAccept;
-        }
-      } else {
-        peer.sendPacket(new SwapErrorPacket({
-          r_hash,
-          errorMessage: SwapFailureReason[SwapFailureReason.OrderUnavailable],
-        }, requestPacket.header.id));
+    if (!order) {
+      peer.sendPacket(new SwapErrorPacket({
+        r_hash,
+        errorMessage: SwapFailureReason[SwapFailureReason.OrderNotFound],
+      }, requestPacket.header.id));
+      return;
+    }
+
+    const availableQuantity = order.hold ? order.quantity - order.hold : order.quantity;
+    // TODO: accept the smaller of the proposed quantity and the available quantity
+    if (availableQuantity >= proposedQuantity) {
+      // put accepted quantity on hold
+      // quantityToAccept = Math.min(proposedQuantity, availableQuantity);
+      /** The quantity of the order that we will accept */
+      const quantity = proposedQuantity;
+
+      order.hold = order.hold ? order.hold + quantity : quantity;
+
+      // try to accept the deal
+      const orderToAccept = {
+        quantity,
+        localId: order.localId,
+        price: order.price,
+        isBuy: order.isBuy,
+      };
+      const dealAccepted = await this.swaps!.acceptDeal(orderToAccept, requestPacket, peer);
+      if (!dealAccepted) {
+        // release hold amount and reject swap
+        order.hold -= quantity;
       }
     } else {
       peer.sendPacket(new SwapErrorPacket({
         r_hash,
-        errorMessage: SwapFailureReason[SwapFailureReason.OrderNotFound],
+        errorMessage: SwapFailureReason[SwapFailureReason.OrderUnavailable],
```

Comment:
`OrderUnavailable ` - this name and the attached description does not reflect the situation properly.

Reply:
How do you figure? My thinking is that the order is not currently available because it is on hold - although the order exists. Would `OrderOnHold` make more sense?

Reply:
I'll go ahead and make this `OrderOnHold` because I think it's a more specific indication of why the swap was rejected.